calculus.py

```diff
@@ -256,6 +256,54 @@ def trapezoid_scipy(func, l_lim, u_lim, steps=10000):
     integral_value = sp.integrate.trapezoid(y, x)   # calculate the integral using numpy
     return integral_value
 
+def trapezoid_python(func, l_lim, u_lim, steps=10000):
+    '''
+    This function implements trapezoidal rule by a pure python implementation
+    by evaluating the integral of the input function over the limits given, and
+    in the input number of steps and gives as output the integral value in numpy 
+    floating point decimal. If the input function is infinite at any point that point
+    is modified to a slightly higher value.
+
+    Parameters:
+    - func: integrand (could be a custom defined function or a standard function like np.sin)
+    - l_lim: lower limit of integration
+    - u_lim: upper limit of integration
+    - steps: number of steps (default value 10000)
+
+    Returns:
+    - integral of the input function using numpy.trapezoid function
+
+    The plotting functionality can be incorporated just like in trapezoid_numpy()
+    or trapezoid_scipy()
+    '''
+
+    # check if the integrand is infinite at lower limit, if yes slightly change the limit
+    try:
```

Comment:
I like the error handling on the first step of the function, good way to prevent nesting. Not a request for a change or anything, just a comment.